ethereum/core/src/main/java/tech/pegasys/pantheon/ethereum/core/ReadOnlyMutableAccount.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright 2019 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.pantheon.ethereum.core;
+
+import tech.pegasys.pantheon.util.bytes.Bytes32;
+import tech.pegasys.pantheon.util.bytes.BytesValue;
+import tech.pegasys.pantheon.util.uint.UInt256;
+
+import java.util.Map;
+import java.util.NavigableMap;
+
+public class ReadOnlyMutableAccount implements MutableAccount {
+
+  private MutableAccount mutableAccount;
+
+  public ReadOnlyMutableAccount(final MutableAccount publicAccount) {
+    this.mutableAccount = publicAccount;
+  }
+
+  @Override
+  public void setNonce(final long value) {
+    throw new UnsupportedOperationException();
```

Comment:
What are the implications of this if a new OPCode is added that we don't private contracts to call in the public state?

Reply:
We get into quite a nasty situation where pantheon tries to re-evaluate this over and over.